app/models/experimentation/notifications_experiment.rb

```diff
@@ -69,13 +71,11 @@ def record_notification_results
       # TODO: Look at query performance
       time(__method__) do
         treatment_group_memberships
-          .joins(treatment_group: :reminder_templates)
+          .joins(:patient, treatment_group: :reminder_templates)
```

Comment:
This is the main change. The rest is to support timing class methods.

Reply:
looks reasonable -- wanna remove the TODO above?

Reply:
What is the measured difference in performance?

Reply:
For 200 patients, this takes 2.5s locally instead of 3.5s, because we aren't making those many extra queries. The difference will scale linearly with the number of patients.

I would like to revisit the overall performance of this query, this only improves the narrow fix I shipped yesterday morning.